inst/tests/tests.Rraw

```diff
@@ -7039,6 +7039,11 @@ test(1521, x[, b := 5], data.table(a=c(1,2), b=5))
 
 # Fix for #1160, fastmean retaining attributes
 x = data.table(a = c(2,2,1,1,2), b=setattr(1:5, 'class', c('bla', 'integer')))
```

Comment:
This test is building explicitly wrong (IMO) behavior by expecting `fastmean` to run on `bla`-class objects.

Essentially it's forcing `mean.default` to be run on all classes & then returning the object with the old class... As we've seen in #3533 etc, this implicit bypassing of the method dispatch can be ill-advised.

IIUC that's why `mean.default` will coerce to `numeric` instead of retaining `c('bla', 'integer')` -- especially the `integer` part feels explicitly wrong. 

```
x = data.table(a = c(2,2,1,1,2), b=1:5)
setattr(x$b, 'class', c('bla', 'integer'))
class(mean(x$b))
# [1] numeric
```

The original issue of #1160 would have been fixed by this PR as well (just not using `fastmean`), so leaving the test in and adjusting by providing a `mean.bla` method. @arunsrinivasan made the original PR, any thoughts?

Reply:
only that there is one pending issue related to fast mean https://github.com/Rdatatable/data.table/issues/2799